[CI] DocumentSubsetBitsetCacheTests testCacheUnderConcurrentAccess failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/o6htlrcdwee6c/tests/:x-pack:plugin:core:test/org.elasticsearch.xpack.core.security.authz.accesscontrol.DocumentSubsetBitsetCacheTests/testCacheUnderConcurrentAccess

**Reproduction line:**
`./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.security.authz.accesscontrol.DocumentSubsetBitsetCacheTests.testCacheUnderConcurrentAccess" -Dtests.seed=E0229A8C30B449CA -Dtests.locale=sr-Latn-RS -Dtests.timezone=Etc/GMT0 -Druntime.java=17`

**Applicable branches:**
8.5

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.core.security.authz.accesscontrol.DocumentSubsetBitsetCacheTests&tests.test=testCacheUnderConcurrentAccess

**Failure excerpt:**
```
java.lang.AssertionError: Query threads did not complete in expected time

  at __randomizedtesting.SeedInfo.seed([E0229A8C30B449CA:C28FE35AF1C8E98]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.assertTrue(Assert.java:41)
  at org.elasticsearch.xpack.core.security.authz.accesscontrol.DocumentSubsetBitsetCacheTests.lambda$testCacheUnderConcurrentAccess$13(DocumentSubsetBitsetCacheTests.java:418)
  at org.elasticsearch.xpack.core.security.authz.accesscontrol.DocumentSubsetBitsetCacheTests.runTestOnIndices(DocumentSubsetBitsetCacheTests.java:649)
  at org.elasticsearch.xpack.core.security.authz.accesscontrol.DocumentSubsetBitsetCacheTests.testCacheUnderConcurrentAccess(DocumentSubsetBitsetCacheTests.java:394)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/DocumentSubsetBitsetCacheTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/DocumentSubsetBitsetCacheTests.java
@@ -359,13 +359,12 @@ public class DocumentSubsetBitsetCacheTests extends ESTestCase {
         });
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/91471")
     public void testCacheUnderConcurrentAccess() throws Exception {
         // This value is based on the internal implementation details of lucene's FixedBitSet
         // If the implementation changes, this can be safely updated to match the new ram usage for a single bitset
         final long expectedBytesPerBitSet = 56;
 
-        final int concurrentThreads = randomIntBetween(5, 15);
+        final int concurrentThreads = randomIntBetween(5, 8);
         final int numberOfIndices = randomIntBetween(3, 8);
 
         // Force cache evictions by setting the size to be less than the number of distinct queries we search on.
@@ -396,27 +395,39 @@ public class DocumentSubsetBitsetCacheTests extends ESTestCase {
                 final CountDownLatch start = new CountDownLatch(concurrentThreads);
                 final CountDownLatch end = new CountDownLatch(concurrentThreads);
                 final Set<BitSet> uniqueBitSets = Collections.synchronizedSet(Collections.newSetFromMap(new IdentityHashMap<>()));
+                final AtomicReference<Throwable> exceptionInThread = new AtomicReference<>();
                 for (int thread = 0; thread < concurrentThreads; thread++) {
                     threads.submit(() -> {
-                        start.countDown();
-                        start.await(100, TimeUnit.MILLISECONDS);
-                        for (int loop = 0; loop < 15; loop++) {
-                            for (int field = 1; field <= FIELD_COUNT; field++) {
-                                final TermQueryBuilder queryBuilder = QueryBuilders.termQuery("field-" + field, "value-" + field);
-                                final TestIndexContext randomContext = randomFrom(contexts);
-                                final Query query = queryBuilder.toQuery(randomContext.searchExecutionContext);
-                                final BitSet bitSet = cache.getBitSet(query, randomContext.leafReaderContext);
-                                assertThat(bitSet, notNullValue());
-                                assertThat(bitSet.ramBytesUsed(), equalTo(expectedBytesPerBitSet));
-                                uniqueBitSets.add(bitSet);
+                        try {
+                            start.countDown();
+                            if (false == start.await(100, TimeUnit.MILLISECONDS)) {
+                                // We still proceed even when some threads are not ready. All threads being ready increases the chance
+                                // of them running concurrently and competing for caching. But this is not guaranteed either way.
+                                logger.info("[{}] out of [{}] worker threads are ready", start.getCount(), concurrentThreads);
                             }
+                            for (int loop = 0; loop < 5; loop++) {
+                                for (int field = 1; field <= FIELD_COUNT; field++) {
+                                    final TermQueryBuilder queryBuilder = QueryBuilders.termQuery("field-" + field, "value-" + field);
+                                    final TestIndexContext randomContext = randomFrom(contexts);
+                                    final Query query = queryBuilder.toQuery(randomContext.searchExecutionContext);
+                                    final BitSet bitSet = cache.getBitSet(query, randomContext.leafReaderContext);
+                                    assertThat(bitSet, notNullValue());
+                                    assertThat(bitSet.ramBytesUsed(), equalTo(expectedBytesPerBitSet));
+                                    uniqueBitSets.add(bitSet);
+                                }
+                            }
+                            end.countDown();
+                        } catch (Throwable e) {
+                            logger.warn("caught exception in worker thread", e);
+                            exceptionInThread.compareAndSet(null, e);
                         }
-                        end.countDown();
                         return null;
                     });
                 }
 
-                assertTrue("Query threads did not complete in expected time", end.await(1, TimeUnit.SECONDS));
+                if (false == end.await(1, TimeUnit.SECONDS)) {
+                    fail("Query threads did not complete in expected time. Possible exception [" + exceptionInThread.get() + "]");
+                }
 
                 threads.shutdown();
                 assertTrue("Cleanup thread did not complete in expected time", threads.awaitTermination(3, TimeUnit.SECONDS));

